File: problem_0002/problem_0002.m
function output = F(N)
  if N<3
     output = 1;
     break
  end
  a = [1,1];
  for i=3:N
    a = [a sum(a)];
    a = a(2:3);
  end
  output = a(2);
end

S=v=i=0;
while v<4*10**6
  i += 1;
  v = F(i);
  if mod(v,2)==0
    S += v;
  end
end

disp(S)

File: problem_0002/problem_0002.r
F <- function(N){
  if (N<3){return(1)}
  a = c(1,1)
  for (n in c(3:N)){
    a = append(a, sum(a))
    a = a[2:3]
  }
  return(a[2])
}

S = i = 0
while(TRUE){
  v = F(i)
  if (v>4*10**6){break}
  if (v%%2==0){S=S+v}
  i = i+1
}
print(S)
